mam/src/mam.rs

```diff
@@ -4,27 +4,25 @@ use sign::iss;
 use merkle;
 use trytes::*;
 use tmath::*;
-use auth::*;
 use mask::*;
 use errors::*;
+use pascal;
 
 /// generate the message key for a root and an index.
 /// It copies `root` to `key`, and adds `index` to it.
-pub fn message_key(root: &[Trit], index: usize, key: &mut [Trit]) -> Result<usize, MamError> {
-    if key.len() < root.len() {
-        Err(MamError::ArrayOutOfBounds)
-    } else {
-        key.clone_from_slice(root);
-        add_assign(&mut key[..HASH_LENGTH], index as isize);
-        Ok(key.len())
-    }
+pub fn message_key<C: Curl<Trit>>(side_key: &[Trit], root: &[Trit], index: usize, curl: &mut C) {
```

Comment:
Please update the documentation.

Reply:
Also, could we flip the order of root & side_key?

Reply:
You think that root should be absorbed first?